core/dom/domobject.js

```diff
@@ -257,7 +264,22 @@ CKEDITOR.dom.domObject.prototype = ( function() {
 	 * @returns {Number} A unique ID.
 	 */
 	domObjectProto.getUniqueId = function() {
-		return this.$[ 'data-cke-expando' ] || ( this.$[ 'data-cke-expando' ] = CKEDITOR.tools.getNextNumber() );
+		var expandoNumber;
+
+		try {
+			expandoNumber = this.$[ 'data-cke-expando' ] || ( this.$[ 'data-cke-expando' ] = CKEDITOR.tools.getNextNumber() );
+
+			// Editable is cached unlike other elements, so we can use it to store expando number.
+			// We need it to properly cleanup custom data in case of permission denied
+			// thrown by Edge when accessing native element of detached editable (#3115).
+			if ( this instanceof CKEDITOR.editable ) {
+				this._.expandoNumber = expandoNumber;
+			}
+		} catch ( e ) {
+			expandoNumber = this._ && this._.expandoNumber;
```

Comment:
Could you make sure that this logic is actually cleaning up custom data by unit tests?

Reply:
We can't check if custom data is cleaned as custom data object is private. I'll add tests to `editable.getUniqueId` to check if id is returned regardless of error.